test cases/common/161 dolphin option/meson.build

```diff
@@ -0,0 +1,12 @@
+project('dolphin option', 'c')
+
+x264_dep = dependency('x264', options : ['video', 'patented'])
+unfound_dep = dependency('notfound', required : false)
+
+exe = executable('encprog', 'encprog.c',
+  dependencies : x264_dep)
+test('enc test', exe)
+
+subdir('nogo', if_found : x264_dep)
+subdir('nogo', if_found : unfound_dep)
```

Comment:
Instead of doing `subdir()` twice, why not check for zlib, set options via a unit test (that is only run when zlib is found) and test various use-cases? For instance, we need to check things like:

1. Using a disabler should cascade that change down to everything else that uses its children. Example: custom target that uses an executable, which is in turn used in a library, and then another executable, and then a test.
2. Tests should not be defined if they use a disabler
3. Does the `options:` kwarg do an OR on the options or an AND?
4. How does setting an option to `auto` behave?
etc.